src/osp/universe/universe.h

```diff
@@ -124,18 +88,19 @@ struct CoSpaceTransform
     // Ignore and use m_parentSat's position and rotation instead if non-null
     Quaterniond     m_rotation;
     Vector3g        m_position;
+
     int             m_precision{10}; // 1 meter = 2^m_precision
 };
 
-
 struct CoSpaceSatData
 {
     uint32_t        m_satCount;
     uint32_t        m_satCapacity;
 
     Corrade::Containers::Array<unsigned char> m_data;
     SatVectorDesc<spaceint_t, 3>    m_satPositions;
-    SatVectorDesc<float, 3>         m_satVelocities;
+    SatVectorDesc<double, 3>        m_satVelocities;
+    SatVectorDesc<double, 4>        m_satRotations;
```

Comment:
Variable 'osp::universe::CoSpaceSatData::m_satRotations' is uninitialized. Always initialize a member variable (type.6).

Warning
Variable 'osp::universe::CoSpaceSatData::m_satRotations' is uninitialized. Always initialize a member variable (type.6). 






Not clear to me if this is valid for a complex type like this.

Wonder if we need to adjust rules.

Reply:
If you didn't know, you can view these automated code scanning reports by looking at the Files Changed tab.